blockchain/blockdao/grpcblockdao.go

```diff
@@ -57,8 +59,10 @@ func (gbd *GrpcBlockDAO) Start(ctx context.Context) error {
 	opts := []grpc.DialOption{}
 	if gbd.insecure {
 		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	} else {
+		opts = append(opts, grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{MinVersion: tls.VersionTLS12})))
 	}
-	gbd.conn, err = grpc.Dial(gbd.url, opts...)
+	gbd.conn, err = grpc.NewClient(gbd.url, opts...)
```

Comment:
IMO since the blockdao data (blocks, transactions) are public data of the chain, there is no privacy/protection need. so we can just keep using insecure, no need to use TLS/encryption. Is that right or there's something else to consider?